### ISSUE TITLE
Github.java:37-39: Complete Github API methods. When...

### ISSUE THREAD

**Issue by 0pdd** [decorators-squad/versioneye-api#112]
The puzzle `101-27d97a0d` in [`src/main/java/com/amihaiemil/versioneye/Github.java`](https://github.com/decorators-squad/versioneye-api/blob/master/src/main/java/com/amihaiemil/versioneye/Github.java) (lines 37-39) has to be resolved: "Complete Github API methods. When possible, think that you can implement those methods on a Project. (e.g. see Project.hook())."

The puzzle was created by amihaiemil on 06-Jun-17. 

Estimate: 30 minutes, role: DEV.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0pdd** [comment]
@amihaiemil please pay attention to this new issue.

**Comment by SherifWaly** [comment]
@amihaiemil I want to do this one.
I want to ask whether this would be a good design for the methods or not.
for GET, POST and DELETE /github/{repo_key} I think they should be implemented at ``Repositories`` as GET returns a more detailed information about the repository and I should make a class called for example ``DetailedRepository``.

What do you think about that :) ?

**Comment by amihaiemil** [comment]
@SherifWaly hmmm... I think 

1) DELETE should go in the Repository interface. So once the usre has a Repository, he can simply call ``delete()`` on it, without needing to get its id -- same as for a Project

2) GET /github/{repo_key} can go in ``Repositories``, but do not make another class. The returned JSON will be animated by the existing implementation, which won't have all the needed methods, but it has the ``json()`` method which returns the raw content, with which the user can play. I'm affraid that having yet another class like ``DetailedRepository`` will be confusing.

3) POST /github/{repo_key} is an endpoint that I do not understand: why does it need to take a File input? If I give the repo coordinates and the token which has at least read access on Github, can't VersionEye look for it in the repository? It seems pointless, and I would leave this endpoint out for version 1... anyway, for file upload with ``jcabi-http`` we have to wait for a ticket to be resolved there, or find another way on our own :D 

So, for now, only DELETE and GET. 
Makes sense? What do you say?

**Comment by amihaiemil** [comment]
By the way, I'm thinking to release the first version soon, in the next weeks - it has a good shape already, I think. Mock version is not finished, of course, but that can wait until later versions.

**Comment by SherifWaly** [comment]
@amihaiemil Ok, let's start with DELETE and GET :D 
I also forgot the /github/hook it's implemented in Project but I think it should be done here also, what do you think :D ?

**Comment by amihaiemil** [comment]
@SherifWaly I think it's perfect in the Project. Since no matter what, the user will have to fetch a Project first and get its id, and only after that, to look for the method in this interface... it's more procedural. 

Like it's done now, he gets an instance of Project and sees "ok, I can hook THIS project, to Github".
Again, otherwise, he would have to do more steps, to get the same outcome.

**Comment by SherifWaly** [comment]
@amihaiemil Ok, so i will implement DELETE and GET methods

**Comment by amihaiemil** [comment]
@SherifWaly don't forget about PDD. I think the DELETE should work in 30 min, not sure if you have time for both. Puzzles :D

**Comment by 0pdd** [comment]
The puzzle `101-27d97a0d` has disappeared from the source code, that's why I closed this issue. //cc @amihaiemil

**Comment by amihaiemil** [comment]
@SherifWaly done here, thanks :)
